source/common/config/subscription_factory_impl.cc

```diff
@@ -62,8 +63,15 @@ absl::StatusOr<SubscriptionPtr> SubscriptionFactoryImpl::subscriptionFromConfigS
   }
   case envoy::config::core::v3::ConfigSource::ConfigSourceSpecifierCase::kApiConfigSource: {
```

Comment:
What's the difference between this function and `SubscriptionFactoryImpl::collectionSubscriptionFromUrl`, which has a similar check and isn't changed? Is it because secrets are currently not fetched as collections?

Reply:
`SubscriptionFactoryImpl::collectionSubscriptionFromUrl` is for resource locator based subscriptions which is not supported by SDS. See https://github.com/envoyproxy/envoy/blob/main/source/common/secret/sds_api.cc#L35. LDS supports it https://github.com/envoyproxy/envoy/blob/main/source/common/listener_manager/lds_api.cc#L34 for example.